compiler/src/dmd/errors.d

```diff
@@ -929,3 +941,54 @@ private void writeHighlights(Console con, ref const OutBuffer buf)
             fputc(c, con.fp);
     }
 }
+
+/************************************************************
+ * This is a simple scheme to eliminate duplication of error
+ * messages, which we'll define has having the same location.
+ * The complication is that messages come in groups,
+ * all with the same location. This is dealt with by having
+ * some messages trigger the start of a group, which ends only
+ * when another start is seen, at wich point the previous location
+ * is logged in an associative array. The contents of the associative
+ * array is consulted when writing out a message.
+ */
+
+private struct MessageGroup
+{
+  nothrow:
+
+    /*********
+     * The start of a new group.
+     * Params:
+     *  loc = location associated with the mssage
+     */
+    void start(ref const Loc loc)
+    {
+        if (loc.linnum() == 0)      // print messages with unknown location
+            suppress = false;
```

Comment:
The style guidelines dictate that if one of the branches has braces, the other one should too.